src/components/street-geo.js

```diff
@@ -0,0 +1,106 @@
+/* global AFRAME, THREE */
+const MAPBOX_ACCESS_TOKEN_VALUE = 'pk.eyJ1Ijoia2llcmFuZmFyciIsImEiOiJjazB0NWh2YncwOW9rM25sd2p0YTlxemk2In0.mLl4sNGDFbz_QXk0GIK02Q';
+const GOOGLE_API_KEY = 'AIzaSyAQshwLVKTpwTfPJxFEkEzOdP_cgmixTCQ';
+
+
+AFRAME.registerComponent('street-geo', {
+  schema: {
+    longitude: { type: 'number', default: 0 },
+    latitude: { type: 'number', default: 0 },
+    elevation: { type: 'number', default: 0 },
+    maps: { type: 'array', default: [] }
+  },
+  init: function () {
+  	/* 
+	  	Function names for the given function types must have the following format:
+	  	create function: <mapType>Create,
+	  	update function: <mapType>Update,
+  	*/
+    this.mapTypes = ['mapbox2d', 'google3d'];  
+    this.elevationHeightConstant = 32.49158;	
+  },
+  update: function (oldData) {
+    const data = this.data;
+    const el = this.el;
+
+    const updatedData = AFRAME.utils.diff(oldData, data);
+
+    for (const mapType of this.mapTypes) {
+    	// create map function with name: <mapType>Create
+    	const createElementFunction = this[mapType + 'Create'].bind(this);
+	    // create Map element and save a link to it in this[mapType]
+	    if (data.maps.includes(mapType) && !this[mapType]) {
+	    	this[mapType] = createElementFunction();
+	    } else if (data.maps.includes(mapType) && (updatedData.longitude || updatedData.latitude || updatedData.elevation)) {
+	    	// call update map function with name: <mapType>Update
+	    	this[mapType + 'Update'].bind(this)();
+	    } else if (this[mapType] && !data.maps.includes(mapType)) {
+	    	// remove element from DOM and from this object
+	    	this.el.removeChild(this[mapType]);
+	    	this[mapType] = null;
+	    }    	
+    }
+  },
+  mapbox2dCreate: function () {
+    const data = this.data;
+    const el = this.el;
+
+		const mapbox2dElement = document.createElement('a-entity');
+		mapbox2dElement.setAttribute('data-layer-name', 'Mapbox Satellite Streets');
+		mapbox2dElement.setAttribute('geometry', 'primitive: plane; width: 512; height: 512;');
```

Comment:
You have a mix of spaces and tabs in the file, be sure to replace all tabs by spaces.